CleverGirl/CleverGirl/Helper/AEHelper.cs

```diff
@@ -212,295 +177,87 @@ public static bool MeleeDamageOutweighsRisk(float attackerMeleeDam, Mech attacke
+                        Mod.Log.Debug?.Write($" Skipping ammoMode {ammoMode} for {wep.UIName} in ranged set as distance: {distance} < minRange: {wep.MinRange}");
                         continue;
                     }
-                }
-
-                if (!wep.CanFire)
-                {
-                    Mod.Log.Debug?.Write($" Skipping {wep.UIName} in ranged set it cannot fire (out of ammo, damaged, etc)");
-                    continue;
-                }
 
-                weaponsToFire.Add(cWeap);
-            }
-
-            return MakeWeaponSets(weaponsToFire);
-
-        }
-
-
-
-        // === CLONE METHODS BELOW ==
-
-        // CLONE OF HBS CODE - LIKELY BRITTLE!
-        public static List<List<CondensedWeapon>> MakeWeaponSets(List<CondensedWeapon> potentialWeapons)
-        {
-            List<List<CondensedWeapon>> list = new List<List<CondensedWeapon>>();
-            if (potentialWeapons.Count > 0)
-            {
-                CondensedWeapon item = potentialWeapons[0];
-                List<CondensedWeapon> range = potentialWeapons.GetRange(1, potentialWeapons.Count - 1);
-                List<List<CondensedWeapon>> list2 = MakeWeaponSets(range);
-                for (int i = 0; i < list2.Count; i++)
-                {
-                    List<CondensedWeapon> list3 = list2[i];
-                    list.Add(list3);
-                    list.Add(new List<CondensedWeapon>(list3) { item });
-                }
-            }
-            else
-            {
-                List<CondensedWeapon> item2 = new List<CondensedWeapon>();
-                list.Add(item2);
-            }
-            return list;
-        }
-
-        // CLONE OF HBS CODE - LIKELY BRITTLE!
-        public static List<List<CondensedWeapon>> MakeWeaponSetsForEvasive(List<CondensedWeapon> potentialWeapons, float toHitFrac, ICombatant target, Vector3 shooterPosition)
-        {
-            List<CondensedWeapon> likelyToHitWeapons = new List<CondensedWeapon>();
-            List<CondensedWeapon> unlikelyNonAmmoWeapons = new List<CondensedWeapon>();
-            List<CondensedWeapon> unlikelyAmmoWeapons = new List<CondensedWeapon>();
-
-            // Separate weapons into likely to hit, and unlikely to hit. Add only a single unlikely to hit weapon to the sets to be created.
-            // TODO: Make this multi-step marginal... don't fire 9% weapons, but do fire 30%?
-            for (int i = 0; i < potentialWeapons.Count; i++)
-            {
-                CondensedWeapon weapon = potentialWeapons[i];
-                if (weapon.First.CanFire)
-                {
-                    float toHitFromPosition = weapon.First.GetToHitFromPosition(target, 1, shooterPosition, target.CurrentPosition, true, true, false);
-                    if (toHitFromPosition < toHitFrac)
+                    // Check one-shot weapons for accuracy
+                    if (wep.weaponDef.StartingAmmoCapacity == wep.weaponDef.ShotsWhenFired)
```

Comment:
I believe BTA has a 'reloading' One-Shot weapons, which is why the prior logic existed. You should check with BD that this change will continue to account for those weapons.

Reply:
Discussed it with BTA crew and decided to switch to a more in-depth check using CAC features since `StartingAmmoCapacity` is not reliable as definition of internal ammunition.